stan/math/prim/fun/select.hpp

```diff
@@ -0,0 +1,176 @@
+#ifndef STAN_MATH_PRIM_FUN_SELECT_HPP
+#define STAN_MATH_PRIM_FUN_SELECT_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/functor/apply_scalar_binary.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return the second argument if the first argument is true
+ * and otherwise return the third argument.
+ *
+ * <code>select(c, y1, y0) = c ? y1 : y0</code>.
```

Comment:
(optional) I just prefer using ticks here \` doxygen is able to parse that like it's markdown